src/Concerns/HasLayouts.php

```diff
@@ -2,58 +2,166 @@
 
 namespace Plank\Contentable\Concerns;
 
+use Illuminate\Contracts\Database\Eloquent\Builder;
+use Illuminate\Database\Eloquent\Collection;
 use Illuminate\Database\Eloquent\Model;
-use Plank\Contentable\Contracts\Layout as LayoutContract;
+use Illuminate\Database\Eloquent\Relations\BelongsTo;
+use Plank\Contentable\Contracts\Layout;
 use Plank\Contentable\Contracts\Layoutable;
+use Plank\Contentable\Enums\LayoutMode;
+use Plank\Contentable\Enums\LayoutType;
+use Plank\Contentable\Exceptions\MissingLayoutException;
 
 /**
  * @mixin Model
  * @mixin Layoutable
  */
 trait HasLayouts
 {
+    public function layout(): Layout
+    {
+        if ($this->relatedLayout === null) {
+            throw MissingLayoutException::show(static::class, $this->getKey());
+        }
+
+        return $this->relatedLayout;
+    }
+
+    public static function indexLayout(): Layout
+    {
+        $layoutModel = static::layoutModel();
+
+        $layout = $layoutModel::query()
+            ->where($layoutModel::getLayoutKeyColumn(), static::indexLayoutKey())
+            ->first();
+
+        if ($layout === null) {
+            throw MissingLayoutException::index(static::class);
+        }
+
+        return $layout;
+    }
+
+    public function layouts(): Collection
+    {
+        $layoutModel = static::layoutModel();
+
+        return $layoutModel::query()
+            ->where($layoutModel::getLayoutableColumn(), static::layoutKey())
+            ->when($this->globalLayouts(), function (Builder $query) use ($layoutModel) {
+                $query->orWhere(function (Builder $query) use ($layoutModel) {
+                    $query->where($layoutModel::getTypeColumn(), LayoutType::Global)
+                        ->whereNotIn($layoutModel::getLayoutKeyColumn(), $this->excludedLayouts());
+                });
+            })
+            ->get()
+            ->sortBy($layoutModel::getNameColumn());
+    }
+
     /**
-     * {@inheritDoc}
+     * The default implementation ships as a parent relationship
+     *
+     * @return BelongsTo
      */
-    public function layout(): LayoutContract
+    public function relatedLayout()
```

Comment:
Should this have a return type defined in the `: ReturnType` notation to be consistent with the other methods in the trait? 

```suggestion
    public function relatedLayout(): BelongsTo
```

Reply:
I omitted this one intentionally to allow package consumers to change the relationship to whatever they want.